src/app/dynamic_server/DynamicDispatch.cpp

```diff
@@ -0,0 +1,390 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <access/SubjectDescriptor.h>
+#include <app-common/zap-generated/callback.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app-common/zap-generated/ids/Commands.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/GlobalAttributes.h>
+#include <app/MessageDef/AttributeReportIBs.h>
+#include <app/MessageDef/StatusIB.h>
+#include <app/WriteHandler.h>
+#include <app/att-storage.h>
+#include <app/data-model/Decode.h>
+#include <app/util/endpoint-config-api.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/Optional.h>
+#include <lib/core/TLV.h>
+#include <protocols/interaction_model/Constants.h>
+
+/**
+ * This file defines the APIs needed to handle interaction model dispatch.
+ * These are the APIs normally defined in
+ * src/app/util/ember-compatibility-functions.cpp and the generated
+ * IMClusterCommandHandler.cpp but we want a different implementation of these
+ * to enable more dynamic behavior, since not all framework consumers will be
+ * implementing the same server clusters.
+ */
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+
+namespace {
+
+// TODO: Maybe consider making this configurable?  See also
+// CHIPControllerAccessControl.mm.
```

Comment:
Pre-existing, but MTRControllerAccessControl.mm.  But also, this is some of that Matter.framework-specific documentation.

Note that for this file to work right you do need something like MTRControllerAccessControl.mm as well, right?

Reply:
updated, thanks